File: OpenWeatherWebApp/src/app/shared/DTOs/WeatherTypeDTO.ts
import { WeatherConditionDTO } from "./WeatherConditionDTO";

export class WeatherTypeDTO
{
    constructor(
        public  id:number,
        public  main:string,
        public  description:string,
        public  icon:string,
        public  conditionId:number,
        public  condition:WeatherConditionDTO 
    ){}
}

File: OpenWeatherAPI/OpenWeatherAPI.Models/Models/WeatherTypeDTO.cs
using OpenWeatherAPI.Business.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace OpenWeatherAPI.Business.Models
{
    public class WeatherTypeDTO : BaseEntity
    {
        public string Main { get; set; }
        public string Description { get; set; }
        public string Icon { get; set; }
        public int ConditionId { get; set; }

        public WeatherConditionDTO Condition { get; set; }
    }
}
